src/e4c_pk_store_mem.c

```diff
@@ -75,8 +75,13 @@ int e4c_set_id(e4storage *store, const uint8_t *id)
 
 int e4c_set_idseckey(e4storage *store, const uint8_t *key)
 {
+    size_t c2keynotempty = 0;
     memmove(store->privkey, key, sizeof(store->privkey));
     e4c_sync(store);
+    c2keynotempty = zerocheck(store->c2key, sizeof(store->c2key));
```

Comment:
The problem is a chicken and egg one. An end user might set the secret key first, or they might set the c2 key first. In this case, they're setting the device secret key - if we don't have the c2 public key we can't derive a shared point at this stage. If we do, we can and that's what this checks for.

So yes, we'll update the sharedkey if the c2 public key is available, which I'm taking as "non-zero" since we explicitly init all fields to zero at the top.